lib/ansible/modules/packaging/os/yum.py

```diff
@@ -639,6 +639,27 @@ def local_envra(path):
                                header[rpm.RPMTAG_ARCH])
 
 
+def set_env_proxy(conf_file, installroot):
+    yumb = yum_base(conf_file, installroot)
+    namepass = ""
+    schem = "http"
+    try:
+        if yumb.conf.proxy:
+            if 'https:' in yumb.conf.proxy:
+                schem = 'https'
+            else:
+                schem = 'http'
+            if yumb.conf.proxy_username:
+                namepass = namepass + yumb.conf.proxy_username
+                if yumb.conf.proxy_password:
+                    namepass = namepass + ":" + yumb.conf.proxy_password
+                namepass = namepass + '@'
+            os.environ["{0}_proxy".format(schem)] = re.sub(r"({0}://)".format(schem),
+                                                           r"\1" + namepass, yumb.conf.proxy)
+    except AttributeError as ae:
+        raise ae
```

Comment:
Does this work just with the global yum's proxy from `yum.conf` or also with proxies defined in the repo files?